datafusion/sql/src/expr/function.rs

```diff
@@ -59,6 +60,26 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         if let Ok(fun) = BuiltinScalarFunction::from_str(&name) {
             let args =
                 self.function_args_to_expr(function.args, schema, planner_context)?;
+
+            // Translate array_aggregate to aggregate function with array argument.
+            if fun == BuiltinScalarFunction::ArrayAggregate {
```

Comment:
If I understand correctly, data frame use 
https://github.com/apache/arrow-datafusion/blob/eee790f695a58a99e880957d50a33c1f075c8edc/datafusion/expr/src/expr_fn.rs#L153-L161

to call aggregate function. If we need to introduce `AggregateFunction::ArraySum` we just need to use another function that utilizes `ArraySum`, it will have a different path than SQL, right?

Reply:
Let's raise another PR to have a common rewriting rule for SQL, Dataframe, and Expr. I would need more time to deep dive into them.